src/models/service.ts

```diff
@@ -246,4 +252,42 @@ export class ServiceModel extends CosmosdbModelVersioned<
   ): TaskEither<CosmosErrors, Option<RetrievedService>> {
     return super.findLastVersionByModelId([serviceId]);
   }
+
+  public listLastVersionServices(): TaskEither<
+    CosmosErrors,
+    Option<ReadonlyArray<RetrievedService>>
+  > {
+    return tryCatch<CosmosErrors, FeedResponse<RetrievedService>>(
+      () => this.container.items.query(`SELECT * FROM c`).fetchAll(),
+      toCosmosErrorResponse
+    )
+      .map(_ => fromNullable(_.resources))
+      .chain(_ =>
+        _.isSome()
+          ? _.value.length > 0
+            ? fromEither(
+                t
+                  .readonlyArray(this.retrievedItemT)
+                  .decode(_.value)
+                  .map(services =>
+                    some(
+                      Object.values(
+                        services.reduce((prev, curr) => {
+                          const isNewer =
+                            !prev[curr.serviceId] ||
+                            curr.version > prev[curr.serviceId].version;
+                          return {
+                            ...prev,
+                            ...(isNewer ? { [curr.serviceId]: curr } : {})
+                          };
+                        }, {} as Record<string, RetrievedService>)
+                      )
```

Comment:
We may need a named function or a comment line to explain what this code wants to achieve, to avoid readers to go though all implementation details to understand.

Reply:
Comment added in d25a09c